storybookjs/storybook
# Build: Migrate @storybook/scripts to strict-ts

Part of #22176 

## What I did
Migrate @storybook/scripts to strict-ts

More specifically:
- Changed scripts `tsconfig.json` to use strict types
- Created a typechecking script for `@storybook/scripts` based on the one in `prepare/check.ts`
- Resolved type errors that came up after enabling strict types
- Adds `check:scripts` to CI

## How to test
Check types with `cd scripts && yarn check`. The typecheck script should run in CI now as well.

## Checklist

      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers


`["cleanup", "BREAKING CHANGE", "feature request", "bug", "build", "documentation", "maintenance", "dependencies", "other"]`
